subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5641,6 +5643,7 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			return -errno;
 		}
 
+#if defined(CONFIG_MBEDTLS_SERVER_NAME_INDICATION)
```

Comment:
I'm not sure about this, I think the engine should still be able to verify the hostname even if SNI is not used (in case of certificate based security).

Maybe it'd be better to leave the decission to the application whether to set the TLS_HOSTNAME option or not? We could add an additional `bool hostname_verify` field in https://github.com/zephyrproject-rtos/zephyr/blob/main/include/net/lwm2m.h#L140, and then set the socket option if it's set to true?

Reply:
I think that's a good idea or would it be better to introduce a kconfig options which depends on MBEDTLS_X509_CRT_PARSE_C?

Reply:
We don't think we can depend on `MBEDTLS_X509_CRT_PARSE_C` in Kconfig as it does not have a Kconfig counterpart (it's enabled automatically based on other choices).

Reply:
Yes, thatŝ why I edited the command and thought more about the ciphers. But I will do your proposal as it's also possible to have multiple clients which different settings